incubator/hnc/Makefile

```diff
@@ -145,11 +145,9 @@ vet:
 generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths=./api/...
 
-# find or download controller-gen
-# download controller-gen if necessary
-controller-gen: $(CONTROLLER_GEN)
-
-$(CONTROLLER_GEN):
+# Use the version of controller-gen that's checked into vendor/ (see
+# hack/tools.go to see how it got there).
+controller-gen:
 	go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
```

Comment:
Is it that if `CONTROLLER_GEN` (/bin/controller-gen) is not empty on line 150, line 153 won't be run?

Reply:
Oh you will have to view it in the commit diff (the line numbers are for the old makefile).

Reply:
Yes that's exactly it. We're not using makefiles "properly" so that they understand timestamps, and even if we were, timestamps are really brittle and Go compiles really quickly so the best thing for us to do is always rebuild.

Reply:
Thanks for the explanation!